lib/valueflow.cpp

```diff
@@ -1079,6 +1071,38 @@ static void valueFlowArray(TokenList *tokenlist)
     }
 }
 
+static void valueFlowArrayBool(TokenList *tokenlist)
+{
+    for (Token *tok = tokenlist->front(); tok; tok = tok->next()) {
+        if (tok->hasKnownIntValue())
+            continue;
+        const Variable *var = nullptr;
+        bool known = false;
+        std::list<ValueFlow::Value>::const_iterator val =
+            std::find_if(tok->values().begin(), tok->values().end(), std::mem_fn(&ValueFlow::Value::isTokValue));
+        if (val == tok->values().end()) {
+            var = tok->variable();
+            known = true;
+        } else {
+            var = val->tokvalue->variable();
+            known = val->isKnown();
+        }
+        if (!var)
+            continue;
+        if (!var->isArray() || var->isArgument() || var->isStlType())
+            continue;
+        // TODO: Check for function argument
```

Comment:
what could we do with function arguments?

Reply:
We would set them to true as well, but only if its being converted to a bool. We have diagnostics for known function arguments.

Reply:
Yes